### PR TITLE
Convert ancillary workflow components to hooks

### PR DESCRIPTION
**Pull request by zarsky-broad**
An attempt to break up #2422. Best reviewed with whitespace hidden.

### CODE REVIEW

File: src/pages/workspaces/workspace/workflows/WorkflowView.js [hunk 1]
```diff
@@ -480,7 +444,9 @@ const WorkflowView = _.flow(
         !isRedacted ? filterConfigIO(inputsOutputs) : _.identity
       )(config)
 
-      const selectedSnapshotEntityMetadata = modifiedConfig.dataReferenceName ? await Ajax(signal).Workspaces.workspace(namespace, name).snapshotEntityMetadata(namespace, modifiedConfig.dataReferenceName) : undefined
+      const selectedSnapshotEntityMetadata = modifiedConfig.dataReferenceName ?
```
Comment (zarsky-broad) [inline src/pages/workspaces/workspace/workflows/WorkflowView.js:447]: From here down is just Intellij breaking up long lines, which is long overdue for this file, with the exception of the newly renamed `DocumentationCollapse` on line 658.

**Review by rsasch — APPROVED** [review]

File: src/pages/workspaces/workspace/workflows/DataStepContent.js [hunk 2]
```diff
@@ -13,157 +13,145 @@ import { chooseBaseType, chooseRootType, chooseSetType } from 'src/pages/workspa
 import validate from 'validate.js'
 
 
-export default class DataStepContent extends Component {
-  static propTypes = {
```
Comment (breilly2) [inline src/pages/workspaces/workspace/workflows/DataStepContent.js:17]: Huh... I guess I'm not upset about losing this. Funny... I would have felt different a few years ago.

File: src/pages/workspaces/workspace/workflows/WorkflowView.js [hunk 3]
```diff
@@ -266,45 +257,18 @@ const BucketContentModal = ({ workspace: { workspace: { namespace, bucketName }
...
-        ])
-    ])
-  }
+  return div({
+    style: { display: 'flex', margin: '0.5rem 0' },
+    onClick: () => setIsOpened(!isOpened)
+  }, [
+    icon(isOpened ? 'angle-down' : 'angle-right', { style: { marginRight: '0.5rem', color: colors.accent() }, size: 21 }),
+    isOpened ?
+      h(MarkdownViewer, { style: { marginBottom: '1rem' } }, [children]) :
```
Comment (breilly2) [inline src/pages/workspaces/workspace/workflows/WorkflowView.js:269]: `MarkdownViewer`. Nice touch. 👍

**Review by breilly2 — APPROVED** [review]
👍  Code looks good. I tested the selection model in the workspace data tab and everything worked as expected.

**Pull request merged**